src/tools/network.ts

```diff
@@ -44,11 +45,37 @@ const requests = defineTool({
   },
 });
 
-function renderRequest(request: playwright.Request, response: playwright.Response | null) {
+// Only include request/response body for XHR/fetch/json requests
+function shouldIncludeBody(request: playwright.Request): boolean {
+  const resourceType = request.resourceType();
+  if (resourceType === 'xhr' || resourceType === 'fetch')
+    return true;
+  const headers = request.headers();
+  if (headers['x-requested-with']?.toLowerCase() === 'xmlhttprequest')
+    return true;
+  return request.isNavigationRequest() === false &&
```

Comment:
a form submission could be a navigation request, but still has a super interesting body. let's not check for it being a nav request.

Reply:
I fixed the conditions same as the above comment.

- Is application/json request?
- And if there is body, the tool includes it in the result